Source/ServerTrustEvaluation.swift

```diff
@@ -239,53 +290,66 @@ public final class PinnedCertificatesTrustEvaluator: ServerTrustEvaluating {
+    ///   - keys:                     The `SecKey`s to use to validate public keys. Defaults to the public keys of all
+    ///                               certificates included in the main bundle.
+    ///   - performDefaultValidation: Determines whether default validation should be performed in addition to
+    ///                               evaluating the pinned certificates. Defaults to `true`.
+    ///   - validateHost:             Determines whether or not the evaluator should validate the host, in addition to
+    ///                               performing the default evaluation, even if `performDefaultValidation` is `false`.
+    ///                               Defaults to `true`.
     public init(keys: [SecKey] = Bundle.main.publicKeys,
-                validateCertificateChain: Bool = true,
+                performDefaultValidation: Bool = true,
                 validateHost: Bool = true) {
         self.keys = keys
-        self.validateCertificateChain = validateCertificateChain
+        self.performDefaultValidation = performDefaultValidation
         self.validateHost = validateHost
     }
+    
+    public func evaluate(_ trust: SecTrust, forHost host: String) throws -> Bool {
+        guard !keys.isEmpty else {
+            throw AFError.serverTrustEvaluationFailed(reason: .noPublicKeysFound)
+        }
 
-    /// Evaluates the given `SecTrust` value for the given `host`.
-    ///
-    /// - Parameters:
-    ///   - trust: The `SecTrust` value to evaluate.
-    ///   - host:  The host for which to evaluate the `SecTrust` value.
-    /// - Returns: Whether or not the evaluator considers the `SecTrust` value valid for `host`.
-    public func evaluate(_ trust: SecTrust, forHost host: String) -> Bool {
-        let certificateChainEvaluationPassed: Bool = {
-            if validateCertificateChain {
-                let policy = SecPolicyCreateSSL(true, validateHost ? host as CFString : nil)
-                SecTrustSetPolicies(trust, policy)
-
-                return trust.isValid
-            } else {
-                return true
+        if performDefaultValidation {
+            try trust.validate(policy: .default) { (status, result) in
+                AFError.serverTrustEvaluationFailed(reason: .defaultEvaluationFailed(output: .init(host, trust, status, result)))
             }
-        }()
-
-        guard certificateChainEvaluationPassed else { return false }
+        }
+        
+        if validateHost {
+            try trust.validate(policy: .hostname(host)) { (status, result) in
+                AFError.serverTrustEvaluationFailed(reason: .hostValidationFailed(output: .init(host, trust, status, result)))
+            }
+        }
 
-        outerLoop: for serverPublicKey in trust.publicKeys as [AnyHashable] {
-            for pinnedPublicKey in keys as [AnyHashable] {
-                if serverPublicKey == pinnedPublicKey {
-                    return true
+        let keysPinned: Bool = {
```

Comment:
Same comment here about naming. Maybe `keysPinned` to `pinnedKeyFoundInServerKeys`.

Reply:
Updated naming.